src/wx/wxvbam.cpp

```diff
@@ -927,23 +928,26 @@ void MainFrame::SetJoystick()
 {
     /* Remove all attached joysticks to avoid errors while
      * destroying and creating the GameArea `panel`. */
-    joy.Remove();
+    joy.StopPolling();
 
     set_global_accels();
 
     if (!emulating)
         return;
 
-    for (int i = 0; i < 4; i++)
+    std::unordered_set<unsigned> needed_joysticks;
+    for (int i = 0; i < 4; i++) {
```

Comment:
Request to make this "4" a macro like NUM_JOYSTICKS or something, there are other parts of the code that use this value I believe.

Reply:
This is not related to this PR, but feel free to clean up some magic numbers as a separate PR in the core and the core interface, I did a very little bit of that but we need to do a lot more.